[#18512] incomplete list of collectibles

ulisesmac opened the pull request:
fixes #18512

### Summary

This PR shows collectibles having images in its `:collection-data`.
So we are still not listing some collectibles, but the reason is status-go returns them without any image (sometimes just a name).

This PR also adds partial support to display SVG collectibles, but they are still not animated, so this PR is not closing:
-  #18527

### Review notes

Future steps will be align with status-go team to investigate the reason of returning some collectibles empty.


#### Platforms

- Android
- iOS

### Steps to test
<!-- (Specify exact steps to test if there are such) -->

- Open Status
- Navigate to wallet tab
- Add an account owning collectibles
- Check the collectibles listed, we should display more than before, also we should be able to press on them and check their details.

status: ready

ulisesmac committed:
Fix var naming react-native-svg namespace

ulisesmac committed:
Update collectible components to show SVGs

ulisesmac committed:
Display collectibles containing collection images

ulisesmac committed:
Show collectibles containing SVG animations as static images

ulisesmac commented on src/quo/components/profile/collectible/view.cljs:
```diff
@@ -14,12 +16,18 @@
-  [{:keys [style resource size]}]
-  (let [source (if (string? resource) {:uri resource} resource)]
+  [{:keys [style size resource]}]
+  (let [svg? (and (map? resource) (:svg? resource))]
     [rn/view {:style style}
-     [rn/image
-      {:style  (style/tile-style-by-size size)
-       :source source}]]))
+     (if svg?
+       [svg/svg-uri
+        {:uri   (:uri resource)
+         :style (style/tile-style-by-size size)}]
+       [fast-image/fast-image
+        {:style  (style/tile-style-by-size size)
+         :source (if (string? resource)
+                   {:uri resource}
+                   resource)}])]))
```
This component needs to be reimplemented, the collectible component has changed, in the meantime I just added support to show svgs

ulisesmac commented on src/react_native/svg.cljs:
```diff
@@ -6,10 +6,11 @@
 (def path (reagent/adapt-react-class Svg/Path))
 (def rect (reagent/adapt-react-class Svg/Rect))
-(def clippath (reagent/adapt-react-class Svg/ClipPath))
+(def clip-path (reagent/adapt-react-class Svg/ClipPath))
```
Just renamed to properly use kebab-case

ulisesmac commented on src/status_im/contexts/wallet/collectible/view.cljs:
```diff
@@ -92,12 +93,16 @@
-                           :picture     preview-url}}
+                           :picture     (:uri preview-url)}}
          [rn/view {:style style/container}
           [rn/view {:style style/preview-container}
-           [rn/image
-            {:source preview-url
-             :style  style/preview}]]
+           (if (:svg? preview-url)
+             [svg/svg-uri
+              {:uri   (:uri preview-url)
+               :style style/preview}]
+             [rn/image
+              {:source (:uri preview-url)
+               :style  style/preview}])]
```
Add support to display SVGs to the collectible detail page

ulisesmac commented on src/status_im/contexts/wallet/events/collectibles.cljs:
```diff
@@ -24,9 +24,11 @@
 
 (defn displayable-collectible?
   [collectible]
-  (let [{:keys [image-url animation-url]} (:collectible-data collectible)]
+  (let [{{:keys [image-url animation-url]} :collectible-data
+         {collection-image-url :image-url} :collection-data} collectible]
     (or (not (string/blank? animation-url))
-        (not (string/blank? image-url)))))
+        (not (string/blank? image-url))
+        (not (string/blank? collection-image-url)))))
```
Now `displayable-collectible?` is more flexible, it considers the collection images as valid to show

ulisesmac commented on src/status_im/subs/wallet/collectibles.cljs:
```diff
@@ -3,17 +3,31 @@
+  (and (not (string/blank? url))
+       (string/includes? media-type "svg")))
+
 (defn- preview-url
```
Just updated to show the collection image if there's no collectible image

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 6e0f0833 | [#1](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-18577/1/) | 2024-01-20 07:06:39 | ~5 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-240120-070104-6e0f08-pr18577-tests.log) |

smohamedjavid reviewed (approved):
LGTM 🚀 Nice work!

VolodLytvynenko commented:
Unassigned myself. Need to take more old PR instead of this fresh one

VolodLytvynenko commented:
Hey @ulisesmac thank you for PR. Here are a found issues:

## ISSUE 1: [IOS] App crashes and unsmooth scrolling in the collectibles tab

### Steps:
1. Recover a user account with a substantial number of collectibles.
2. Scroll through the collectibles list consistently.

### Actual result:
The app crashes during the process of scrolling collectibles, as observed in the provided screenshot.

https://github.com/status-im/status-mobile/assets/52490791/5d378dff-8ac8-4ecb-973e-53982e042c98

### Expected result:
The scrolling of collectibles should be smooth, and the app should not crash during the scrolling process.

### Logs:
[logs.zip](https://github.com/status-im/status-mobile/files/14051089/logs.zip)

VolodLytvynenko commented:
Unfortunately the issue described in https://github.com/status-im/status-mobile/issues/18613.  
## ISSUE 2: Incomplete list of collectibles

### Steps:
1. Restore the user with collectibles or add his account as watch only

### Actual result:
Collectibles are shown in the Status app
actually, only 5 Mainnet collectibles are fetched 

![image](https://github.com/status-im/status-mobile/assets/52490791/3b7b4213-08e5-44e0-a5bb-e0060488ffd9)

Here is the list of collectibles images that are not fetched: https://opensea.io/assets/ethereum/0x87d598064c736dd0c712d329afcfaa0ccc1921a1/3879 
https://opensea.io/assets/ethereum/0xdf5d68d54433661b1e5e90a547237ffb0adf6ec2/9405 
https://opensea.io/assets/ethereum/0x5d00d312e171be5342067c09bae883f9bcb2003b/33563
https://opensea.io/assets/ethereum/0x4f41d10f7e67fd16bde916b4a6dc3dd101c57394/11525
https://opensea.io/assets/ethereum/0xfaafdc07907ff5120a76b34b731b278c38d6043c/7237005577332284824093637425867105577342876435616966053150409959946886053888
https://opensea.io/assets/ethereum/0x110101156e8f0743948b2a61afcf3994a8fb172e/183
https://opensea.io/assets/ethereum/0x110101156e8f0743948b2a61afcf3994a8fb172e/182
https://opensea.io/assets/ethereum/0x110101156e8f0743948b2a61afcf3994a8fb172e/136
https://opensea.io/assets/ethereum/0x110101156e8f0743948b2a61afcf3994a8fb172e/21

### Expected result:
Collectibles shown in my account using Metamask
13 Mainnet collectibles are shown

https://github.com/status-im/status-mobile/assets/52490791/717a9aba-cf9c-4e80-ae73-39d99281b2f5

VolodLytvynenko commented:
## ISSUE 3: Collectible appears larger than others in list
### Steps:
1. Recover the account as a watch-only account with the address 0x8d2413447ff297d30bdc475f6d5cb00254685aae.
2. Scroll through the list of collectibles.

### Actual Result:
One of the collectibles appears larger than the others in the list.

https://github.com/status-im/status-mobile/assets/52490791/264f32e7-63ff-4ae3-9639-b7359b7b7a7d

### Expected Result:
All collectibles should be displayed at the same size in the collectibles list.

### Logs:
[Status-debug-logs.zip](https://github.com/status-im/status-mobile/files/14055577/Status-debug-logs.zip)

ulisesmac commented:
thanks @VolodLytvynenko I'll solve it

VolodLytvynenko commented:
## ISSUE 4: Some collectibles images are not uploaded

ulisesmac committed:
Improve collectible svg support

ulisesmac committed:
Add support for non svg animations

ulisesmac committed:
Reduce collectibles fetching priority

ulisesmac committed:
Add support to visualize SVGs in collectibles detail

ulisesmac commented:
Hey @VolodLytvynenko 

Thanks for testing.

> ISSUE 1: [IOS] App crashes and unsmooth scrolling in the collectibles tab

on iOS the performance is slower, at least I can only test on a simulator and that's what I noticed. We can't do too much atm, the collectible component will be reworked, so maybe at that point we could check the collectible component performance and improve it.

At least I couldn't replicate a crash with the fixes I did.

> ISSUE 2: Incomplete list of collectibles and some collectibles are differs from what user own

I already asked to wallet desktop team, to check what's happening.

Status go only returns some images, not all :cry: we do the best we can with that info, as soon as it's fixed in status-go, we should see a difference here.

For example:
For this collectible from openSea:
![image](https://github.com/status-im/status-mobile/assets/90291778/826c7f6e-4ea9-4670-bf41-e37cc5c93b3b)

Status-go is just returning the collection image, not the collectible image:
![image](https://github.com/status-im/status-mobile/assets/90291778/b3f9c88b-6d45-483d-8b36-cd0e10536ab6)

so that's what I'm showing, if status-go'd return the collectible, it'd be shown in the mobile app.

Same for some missing collectibles :disappointed: So I'd say let's wait until it's fixed, I'll check with desktop what are the reasons for this to happen.

> ISSUE 3: Collectible appears larger than others in the list

Fixed :smile: 


## Additional notes

- The lightbox is not working for SVG collectibles, I added a temporary alert. the light box would need to be reworked to support these images:
![image](https://github.com/status-im/status-mobile/assets/90291778/0c745ba9-6528-4b15-8b78-c88ff99802bc)

- The banner is showing an `x`  because it also isn't supporting SVGs:
![image](https://github.com/status-im/status-mobile/assets/90291778/f0307cc2-78a2-4a7e-be34-38a0c44ebfe4)


- The first time we get too many collectibles, the keep in the loading state for too long, we need to navigate to other screens because sometimes even if they are already loaded, they are not shown. For the second time you open the app, some of the collectibles were cached, so they will be shown faster.  

  Again, our collectible component is stale, designs have changed, we could do these improvements while implementing the new component.

ulisesmac committed:
lint fix

VolodLytvynenko commented:
@ulisesmac Thank you for clarification and fixes. No issues from my side. PR can be merged

VolodLytvynenko commented:
@ulisesmac I will ask design team about https://github.com/status-im/status-mobile/pull/18577#issuecomment-1910532858. Perhaps this is intentional behavior

VolodLytvynenko commented:
@ulisesmac  On mobile, we should show only collectibles on MVP. Here is a reference to the design answer: https://discord.com/channels/624634427930312714/852533718790570015/1200447531637821531

The PR can be merged, and we can revisit issue 2 only when collectible images fetching is implemented on Status Go. Thank you for investigating this

ulisesmac committed:
Merge branch 'develop' into 18512-incomplete-list-of-collectibles
